target/linux/ramips/image/mt7620.mk

```diff
@@ -7,16 +7,13 @@ define Build/tplink-header
 		-o $@.new -k $@ -r $(IMAGE_ROOTFS) && mv $@.new $@
 endef
 
-define Build/pad-kernel-ex2700
-	cp $@ $@.tmp && dd if=/dev/zero bs=64 count=1 >> $@.tmp \
-		&& dd if=$@.tmp of=$@.new bs=64k conv=sync && truncate -s -64 $@.new \
-		&& cat ex2700-fakeroot.uImage >> $@.new && rm $@.tmp && mv $@.new $@
-endef
-
-define Build/netgear-header
-	$(STAGING_DIR_HOST)/bin/mkdniimg \
-		$(1) -v OpenWrt -i $@ \
-		-o $@.new && mv $@.new $@
+define Build/netgear-kernel
+	let \
+		bsize="$(subst k,*1024,$(1))" \
+		size="$$(stat -c%s '$@')" \
+		size="((1 + ((64 + size) / bsize)) * bsize) - 64"; \
+	dd if=$@ of=$@.new bs=$$size count=1 conv=sync \
```

Comment:
Can't we get rid of the whole block by using the existing [pad-to](https://github.com/lede-project/source/blob/master/include/image-commands.mk#L128) command?

If so, I would suggest to rename ```Build/netgear-kernel``` to ```Build/append-file```:

```
Build/append-file
  cat "$(1)" >> "$@"
endef
```

and change your KERNEL line to something like:

```
KERNEL := $(KERNEL_DTB) | uImage lzma | pad-to 64k | append-file netgear-fakeroot.uImage
```